src/types/name.jl

```diff
@@ -0,0 +1,56 @@
+struct SName
+    region::ShortString15
+    locality1::ShortString15
+    locality2::ShortString15
+end
+
+function Base.print(io::IO, name::SName)
+    print(io, name.region)
+    if !isempty(name.locality1)
+        print(io,"/", name.locality1)
+        if !isempty(name.locality2)
+            print(io,"/", name.locality2)
+        end
+    end
+end
+
+Base.convert(::Type{String}, name::SName) = string(name)
+function Base.convert(::Type{SName}, str::AbstractString)
+    name = try_convert(SName, str)
+    name isa Nothing && DomainError(str, "All timezone name parts must have length < 16")
+    return name
+end
+
+try_convert(::Type{SName}, name::SName) = name
+try_convert(::Type{String}, name::String) = name
+function try_convert(::Type{SName}, str::AbstractString)
+    parts = split(str, "/"; limit=3)
```

Comment:
I'd avoid limiting the `split` here. I'd rather a failure occur here if there are more than 3 parts and not be surprised later in `name_parts`